docs/dogstatsd/internals.md

```diff
@@ -3,66 +3,92 @@
+
 * The buffer is full (contains `dogstatsd_packet_buffer_size, default value: 32`)
+
 or
 
 * A timer is triggered (i.e. `dogstatsd_packer_buffer_flush_timeout, default value: 100ms`)
 
 The PacketBuffer sends it in a Go buffered channel to the worker / parser, meaning that the channels can buffer the Packets on their own while waiting for the worker to read and process them.
 
 In theory, the max memory usage of this Go buffered channel is:
-* packet buffer size * packet size * channel buffer size
-* `dogstatsd_packer_buffer_size` * `dogstatsd_buffer_size` * `dogstatsd_queue_size`
-* 32 * 8192 * 1024 =  256MB
+
+  * packet buffer size * packet size * channel buffer size
+  * `dogstatsd_packer_buffer_size` * `dogstatsd_buffer_size` * `dogstatsd_queue_size`
+  * 32 * 8192 * 1024 =  256MB
 
 ## Worker
 
 Input: slice of Packets
 
 Output: MetricSample sent
 
-The Worker is the part of the Dogstatsd server responsible for parsing the metrics in the bytes array and turning them into MetricSamples. The server spawns multiple workers based on the amount of cores available on the host (the amount of workers created is equal to the number of cores on the machine minus 2. If this result is less than 2, 2 workers are spawned).
+The Worker is the part of the DogStatsD server responsible for parsing the metrics in the bytes array and turning them into MetricSamples.
+
+The server spawns multiple workers based on the amount of cores available on the host:
+
+  * when we are not sharding multiple time sampling pipelines: the amount of workers created is equal to the number of cores on the machine minus 2. If this result is less than 2, 2 workers are spawned.
+  * when we are: the amount of workers is `(number of cores / 2)`.  If this result is less than 2, 2 workers are spawned.
+
 The Worker is using a system called StringInterner to not allocate memory every time a string is needed. Note that this StringInterner is caching a finite number of strings and when it is full it is emptied to start caching strings again. Its size is configurable with `dogstatsd_string_interner_size`.
 
-The MetricSamples created are not directly sent to the Agent aggregator but first to a part called the Batcher.
+The MetricSamples created are not directly sent to the Agent Demultiplexer but first to a part called the Batcher.
 
 ## Batcher
 
 Input: MetricSample from the Worker
 
-Output: slices of MetricSample sent to the aggregator
+Output: slices of MetricSample sent to the Agent Demultiplexer (which is distributing them to the TimeSamplerWorkers)
+
+The role of the Batcher is to accumulate multiple MetricSamples before sending them to the Agent Demultiplexer. Every time it has accumulated 32 MetricSample, it sends them to the Demultiplexer. These 32 MetricSamples are sent in a channel buffering 100 sets, there is one channel per TimeSampler.
```

Comment:
```suggestion
The role of the Batcher is to accumulate multiple MetricSamples before sending them to the Agent Demultiplexer. Every time it accumulates 32 MetricSamples, the Batcher sends them to the Demultiplexer. The Batcher sends 32 MetricSamples in a channel buffering 100 sets. There is one channel per TimeSampler.
```

I rewrote this in a way that conforms to our style guide, but I'm still a bit confused by what it's trying to say. What does the 100 sets refer to? Is it 100 sets of 32 MetricSamples?